witness/witness/cmd/witness/internal/witness/witness.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package witness is designed to make sure the STHs of CT logs are consistent
+// and store/serve/sign them if so.  It is expected that a separate feeder
+// component would be responsible for the actual interaction with logs.
+package witness
```

Comment:
The name of this file reads like "badger mushroom snake"; of the 7 path elements from the root of the repo, 5 of them are `witness`. I strongly urge that the second one should be `ct-witness`. The others I can probably live with, even if it looks a bit bonkers.

Reply:
If you read the comment for the PR, the goal is to eventually have `witness/witness` and `witness/feeder` subdirectories.  So I don't think changing it to `witness/ct-witness` makes much sense?  I could change the top-level one to `ct-witness` if you think that would make things any better.  Otherwise I agree it looks silly but I'm not quite sure of a better structure - suggestions are of course welcome.

Reply:
I missed which repo this was in and thought it was in `trillian-examples`, and the second `witness` directory would sit alongside `ethereum` and `golang`. I retract my suggestion that we need to put `ct` in the name.